homeassistant/config_entries.py

```diff
@@ -396,7 +397,7 @@ def as_dict(self) -> Dict[str, Any]:
             "version": self.version,
             "domain": self.domain,
             "title": self.title,
-            "data": self.data,
+            "data": dict(self.data),
```

Comment:
Does this need to be a `copy.deepcopy`? I don't think `dict()` would be a full solution if someone had nested structures in their `data`?

Reply:
We just need to undo the `MappingProxyType`. When wrapped it doesn't do a deep wrapping. It's not fool proof but catches 90% of bad cases.